converter/src/main/java/gov/cms/qpp/conversion/decode/QrdaDecoderEngine.java

```diff
@@ -165,6 +167,37 @@ private DecodeData decodeChildren(final Element element, final Node parentNode)
 		return decodeData;
 	}
 
+	/**
+	 * Reduces the {@code templateId} {@link Element}s so there are no duplicates.  All other {@link Element}s are left alone.
+	 *
+	 * @param childElements The elements to filter
+	 * @return A {@link List} of {@link Element}s that are filtered.
+	 */
```

Comment:
My guess is that this removes duplicated child nodes from a parent correct? There was a scenario I debugged for the service now issue where one of the template ID's were typo'd. It caused a duplication error where the node had too many aggregate counts... I wonder if this would remove that duplication error and pass a file that is missing the original typo'd template ID.

Reply:
I think perhaps we need to be careful with this.  We don't want to accidentally remove nodes like aggregate counts which should throw a validation error if there's more than one.

Reply:
Not child `Node`s, but child `Element`s, just to be clear.  An element is a duplicate if
- its name is `templateId`, and
- it resolves to the same `TemplateId` enum.

Reply:
Can we try to duplicate the scenario in a test?  Just to be sure that we don't inadvertently remove duplicate aggregate count nodes?

Reply:
Test made.

Reply:
👍